common/src/main/java/com/sedmelluq/discord/lavaplayer/natives/NativeLibLoader.java

```diff
@@ -124,37 +190,63 @@ private static void createDirectoriesWithFullPermissions(Path path) throws IOExc
     }
   }
 
-  private static Architecture detectArchitecture() {
-    String osName = System.getProperty("os.name");
-    String arch = System.getProperty("os.arch");
-    String bits = System.getProperty("sun.arch.data.model");
+  private static class Architecture {
+    private final String osName;
+    private final String archName;
+    private final String libraryPrefix;
+    private final String librarySuffix;
+    private final String systemType;
 
-    if (osName.startsWith("Linux")) {
-      if (arch.startsWith("arm")) {
-        return new Architecture(LINUX_ARM, "lib", ".so");
-      } else if (arch.startsWith("aarch64")) {
-        return new Architecture(LINUX_ARM_64, "lib", ".so");
+    private static String formatSystemName(String osName, String archName) {
+      if (osName == OS_OSX && archName == ARCH_X86_64) {
+        return OS_DARWIN;
       } else {
-        return new Architecture("64".equals(bits) ? LINUX_X86_64 : LINUX_X86, "lib", ".so");
+        return osName + "-" + archName;
       }
-    } else if ((osName.startsWith("Mac") || osName.startsWith("Darwin")) && "64".equals(bits)) {
-      return new Architecture(DARWIN, "lib", ".dylib");
-    } else if (osName.startsWith("Windows") && !osName.startsWith("Windows CE")) {
-      return new Architecture("64".equals(bits) ? WIN_X86_64 : WIN_X86, "", ".dll");
     }
 
-    throw new IllegalStateException("Native libraries not supported on this platform.");
-  }
+    public String formatLibraryName(String libName) {
+      return this.libraryPrefix + libName + this.librarySuffix;
+    }
 
-  private static class Architecture {
-    private final String systemType;
-    private final String libraryPrefix;
-    private final String librarySuffix;
+    public static Architecture detectArchitecture() throws IllegalStateException, IllegalArgumentException {
+      String archName = System.getProperty("lavaplayer.native.arch", archMap.get(SystemUtils.OS_ARCH));
```

Comment:
Commons Lang is a bit of an overkill here, all the constants are either just `System.getProperty` or `someProperty.startsWith` wrappers.